### ISSUE TITLE
add ship:compass, ship:pitch, ship:roll.

### ISSUE THREAD

**Issue by Dunbaratu** [KSP-KOS/KOS#829]
People have been asking about these for a while and we may as well just do them as shown in the title.  It's simple and quick.

compass = navball compass heading.

pitch = angle above horizon (90 - VANG(ship:up,ship:facing:forevector).

roll = angle of deflection between ship:facing:topvector and ship:up, with sign to distinguish left and right.

There are issues with what to return in the case of gimbal locks.  (What is the "compass" and "roll" when going straight up?  There are an infinite number of combinations of compass and roll that would result in the same orientation when going straight up.

**Comment by erendrake** [comment]
is this something we could provide easily with a kerboscript function?

**Comment by Dunbaratu** [comment]
yes but.... I felt as if some of these are silly to do in slow user code.
(i.e. why are SHIP:UP, and SHIP:NORTH first-class citizens, but SHIP:EAST isn't?)

**Comment by space-is-hard** [comment]
Could you return 0 for both roll and compass when abs(pitch) > 89.5? Better yet, have the "bottom" of the ship return the compass heading when the pitch is above that limit, that way the user can orient their rocket to pitch down to that compass heading. It would be useful for the initial stages of a launch.

**Comment by Dunbaratu** [comment]
I'm tempted to have it track a memory of the previous compass heading it had back when it was readable and not gimbal locked yet, and then have it give you the roll that would necessarily go with that heading.  Thus if you pitch up to the point of lock, it assumes you want to "think" in the orientation you were just in a moment ago.

**Comment by erendrake** [comment]
The reason i think it would be interesting to have them in user code it to show that they are simply transformations of the data they already have. how many instructions are we talking about here?

**Comment by Dunbaratu** [comment]
hmm.  well it would be a nice thing to expose as a start of a "standard library".

**Comment by Kemp-J** [comment]
As someone who wasted an afternoon trying to figure out why SHIP:FACING:PITCH was giving nonsense (until I finally found the issue report here), I wholeheartedly agree with having values available that directly relate to the navball readings.

Maybe the documentation should also be updated to reflect that the existing values don't do what might be expected?

**Comment by jonfortescue** [comment]
I think directly exposing the values would be nice, but I echo Kemp's comments on updating the documentation. I think the most critical thing right now would be to clear examples of calculating these values.

**Comment by crupeng-1** [comment]
Is there a way to calculate the ship:compass, because as of yet it isn't implemented in KOS?

**Comment by nuggreat** [comment]
Yes there is see [lib_navball.ks](https://github.com/KSP-KOS/KSLib/blob/master/library/lib_navball.ks) and is associated [documentation](https://github.com/KSP-KOS/KSLib/blob/master/doc/lib_navball.md) and [example ](https://github.com/KSP-KOS/KSLib/blob/master/examples/example_lib_navball.ks).  While there are simpler methods to get these values than the methods employed by lib_navball they are far less general in application.
